Only in ./: build
diff -ru ../launchd-258.22.orig/launchd/src/SystemStarter.c ./launchd/src/SystemStarter.c
--- ../launchd-258.22.orig/launchd/src/SystemStarter.c	2009-07-16 15:30:00.000000000 -0700
+++ ./launchd/src/SystemStarter.c	2009-07-16 15:36:37.000000000 -0700
@@ -152,12 +152,6 @@
 		fwexec(_PATH_BSHELL, "/etc/rc.local", NULL);
 	}
 
-	CFNotificationCenterPostNotificationWithOptions(
-			CFNotificationCenterGetDistributedCenter(),
-			CFSTR("com.apple.startupitems.completed"),
-			NULL, NULL,
-			kCFNotificationDeliverImmediately | kCFNotificationPostToAllSessions);
-
 	r = kevent(kq, NULL, 0, &kev, 1, NULL);
 	assert(r != -1);
 	assert(kev.filter == EVFILT_SIGNAL && kev.ident == SIGTERM);
Only in ./launchd.xcodeproj: wms.pbxuser
