diff --git a/xcodeconfig/libfirehose_kernel.xcconfig b/xcodeconfig/libfirehose_kernel.xcconfig
index f6b2a99..ab26649 100644
--- a/xcodeconfig/libfirehose_kernel.xcconfig
+++ b/xcodeconfig/libfirehose_kernel.xcconfig
@@ -27,7 +27,7 @@ INSTALL_PATH = /usr/local/lib/kernel/
 PRIVATE_HEADERS_FOLDER_PATH = /usr/local/include/kernel/os
 SUPPORTED_PLATFORMS = macosx iphoneos appletvos watchos
 
-HEADER_SEARCH_PATHS = $(PROJECT_DIR) $(SDKROOT)/System/Library/Frameworks/Kernel.framework/PrivateHeaders $(SDKROOT)/System/Library/Frameworks/Kernel.framework/Headers $(SDKROOT)/usr/local/include/os $(SDKROOT)/usr/local/include/firehose
+HEADER_SEARCH_PATHS = $(PROJECT_DIR) /System/Library/Frameworks/Kernel.framework/PrivateHeaders $(SDKROOT)/System/Library/Frameworks/Kernel.framework/Headers /usr/local/include/os /usr/local/include/firehose
 
 GCC_PREPROCESSOR_DEFINITIONS = $(inherited) KERNEL=1 DISPATCH_USE_DTRACE=0
 
